sopel/bot.py

```diff
@@ -200,6 +200,28 @@ def hostmask(self):
 
         return self.users.get(self.nick).hostmask
 
+    def has_channel_privilege(self, channel, privilege):
+        """Tell if the bot has a ``privilege`` level or above in a ``channel``.
+
+        :param str channel: a channel the bot is in
+        :param int privilege: privilege level to check
+        :raise RuntimeError: when the channel is unknown
```

Comment:
I'm not sold on this exception type. Passing in a channel without knowing if the bot is in it first is a caller error. Wouldn't `ValueError` make sense?

Reply:
It's kind of a generic `Exception` tbh:

* `ValueError`: when the value is not in range.
* `TypeError`: when the value is not of the right type
* `RuntimeError`: generic error when something doesn't work at runtime because the dev handle all the cases
* `AssertionError`: a pre or post conditions is invalid

We could have something more accurate, but I'm not sure anyone will bother using it. I'm reluctant to use `assert` (that would raise `AssertionError`, which can be done manually) because a) `assert` can be deactivated; and b) it's not common in the bot's code to do so.

I found few places where it's done, and they confused me (I think it was in Channel and User?), because they are that rare.

Reply:
> * `ValueError`: when the value is not in range

This applies. The expected range of values is "a channel out of the set that the bot knows about", and passing a channel that the bot does _not_ know about is out of range.

About assertions, we… probably should remove those from the `User` and `Channel` objects. They're left over from the initial implementation of user tracking (ea7f14459d3babf1640de504903ee67cd3152878) and I can't imagine anyone getting mad about changing them to proper `raise`s with real exceptions.

Reply:
OK will do that!

Reply:
Fixed that.